### ISSUE TITLE
Test metadata handling

### ISSUE THREAD

**Issue by metan-ucw** [linux-test-project/ltp#308]
As it is the metadata for a testcase are usually written down in the top level comment in the test sources. One of the ideas is to format these in some well defined format so that these can be extracted automatically.

The possible uses are:

* Annotating test failures
* Browseable and searchable document describing our coverage

One of the questions I often get from people is why particular test fails and quite often it's a regression test for a kernel bug/CVE. One of the ideas I do have is to add tags to the tst_test structure so that we can add a link to a kernel commit fixing the issue in the test log in a case that test fails, something as:

```
static struct tst_test test = {
        .min_kver = "2.6.35",
        .needs_root = 1,
        .forks_child = 1,
        .tcnt = ARRAY_SIZE(tcases),
        .setup = setup,
        .cleanup = cleanup,
        .test = do_test
        .tags = &(struct tags) {
                {.id = "kernel_regression", .val = "086e774a57fb"},
                {.id = NULL}
        }
};
```

Then the output may look like:

```
tst_test.c:980: INFO: Timeout per run is 0h 05m 00s
fcntl35.c:98: FAIL: an unprivileged user init the capacity of a pipe to 65536 unexpectedly, expected 4096
fcntl35.c:101: PASS: a privileged user init the capacity of a pipe to 65536 successfully
tst_test.c:432: INFO: Check if the kernel has been patched with https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/commit/?id=086e774a57fb

Summary:
passed   1
failed   1
skipped  0
warnings 0
INFO: Test FAILED
Wed May  2 15:17:35 CEST 2018: fcntl35 FAILED: on 1/1 after 0s
1 fail 0 succeed 1 count 0 internal_error 0 skipped
```

I'm not sure if we should put whole test description into C structures, but at least some tags would make sense for instance for the case of regression tests.

For the actual test description for the purpose of generating coverage document, we may as well reuse some existing format if there is any fitting one and extract the data from the comments with a script.

**Comment by schlad** [comment]
@metan-ucw I'll try to look into this issue.

**Comment by pevik** [comment]
@metan-ucw has prepared metadata in his fork, I'm currently working on docparser (parsing documentation from metadata).

**Comment by pevik** [comment]
First version of this functionality has been merged in docparse/ directory and released in 20210121.
